daemon/network.go

```diff
@@ -267,7 +267,7 @@ func (daemon *Daemon) CreateNetwork(create types.NetworkCreateRequest) (*types.N
 }
 
 func (daemon *Daemon) createNetwork(create types.NetworkCreateRequest, id string, agent bool) (*types.NetworkCreateResponse, error) {
-	if runconfig.IsPreDefinedNetwork(create.Name) && !agent {
+	if runconfig.IsPreDefinedNetwork(create.Name) {
```

Comment:
Sorry for this late comment. Do we really need the `IsPreDefinedNetwork` logic here? I was discussing with Flavio a bit and it seems `daemon.GetNetworkByName` right below should be able to reject all PreDefined networks. This way, we will not even need `IsPreDefinedNetwork` above.

Reply:
Just discussed a bit and we do need separate code paths for Pre-Defined vs User-defined as we do allow name conflicts for the latter category.

Reply:
Thats is what I h added originally. I was reluctant to change.  But we need to be able to differentiate  regular network name exist and predefined network exist .  Review comments were asking me to be more specific to error reports to the callers. Hence I added new type .